controller/gateway/controller_reconciler_utils.go

```diff
@@ -335,7 +336,7 @@ func (r *Reconciler) ensureDataPlaneHasNetworkPolicy(
 func generateDataPlaneNetworkPolicy(
 	namespace string,
 	dataplane *operatorv1beta1.DataPlane,
-	_ *gwtypes.ControlPlane,
+	podLabels map[string]string,
```

Comment:
Let's make sure to add a godoc for this argument as it's not obvious which Pod these labels belong to.